### PR TITLE
[MGDAPI-3827] Tag redis & postgres snapshots on creation

### PR DESCRIPTION
**Pull request by cecobask**
## Overview

Jira: https://issues.redhat.com/browse/MGDAPI-3827

## Verification

- Provision a ROSA cluster using the script in delorean: `make -f make/ocm.mk ocm/sts/cluster/create OCM_ENV=production CLUSTER_NAME=sts-test`
- Setup and wait for cluster admin credentials to become available`rosa create admin --cluster=sts-test`
- Log into the cluster `oc login $(cluster_url) --username cluster-admin --password $(password)`
- Prepare the cluster `make cluster/prepare` using the make commands from the CRO repo
- Setup the sts secret `make setup/sts OCM_ENV=production CLUSTER_NAME=sts-test`
- Run CRO `make run`
- Deploy a redis instance`make cluster/seed/managed/redis`
- Deploy a postgres instance `make cluster/seed/managed/postgres`
- Wait for successful reconciliation
- Create a redis snapshot `make cluster/seed/managed/redissnapshot`
- Create a postgres snapshot `make cluster/seed/managed/postgressnapshot`
- Wait for successful reconciliation
- Ensure the redis and postgres snapshots have been tagged with `red-hat-managed` and other tags
- Trigger deletion of the redis snapshot `oc delete redissnapshot example-redissnapshot -n cloud-resource-operator`
- Trigger deletion of the postgres snapshot `oc delete postgressnapshot example-postgressnapshot -n cloud-resource-operator`
- Wait for successful reconciliation 
- Ensure there are no redis, redissnapshot, postgres, postgressnapshot CRs left in the CRO namespace
- Delete the ROSA STS cluster using the script from delorean: `make -f make/ocm.mk ocm/sts/cluster/delete CLUSTER_NAME=sts-test`

## Checklist
- [ ] This PR includes a change to an instance type, I have used script `hack/<provider>/supported_types.sh` and attached the result below

### CODE REVIEW

**Comment by adam-cattermole** [comment]
Verified:

- New snapshots of both types are created with the appropriate tags when created in an STS enabled ROSA cluster
- Successfully created and deleted postgres/redis/snapshots
- No errors regarding permissions so looks like new policy (integr8ly/delorean#263) is working as intended

/lgtm

File: scripts/sts/create-rhoam-policy.sh [hunk 1]
```diff
@@ -147,16 +146,16 @@ cat <<EOM >"$MINIMAL_POLICY_NAME.json"
...
-                "elasticache:DescribeSnapshots",
-                "elasticache:DescribeUpdateActions",
                 "elasticache:ModifyCacheSubnetGroup",
+                "elasticache:ModifyReplicationGroup",
                 "rds:DeleteDBInstance",
                 "rds:DeleteDBSnapshot",
-                "rds:DeleteDBSubnetGroup"
+                "rds:DeleteDBSubnetGroup",
+                "rds:ModifyDBInstance",
+                "rds:RemoveTagsFromResource"
```
Comment (laurafitzgerald) [inline scripts/sts/create-rhoam-policy.sh:158]: @cecobask do the removals/additions here impact cases where IAM is the authentication method?

Comment (laurafitzgerald) [reply scripts/sts/create-rhoam-policy.sh:158]: actually nvm. I see this is the STS role file not the IAM Credentials Request. 👍

**Pull request merged**